app/api/margatsni/v1/users/auth.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Margatsni
+  module V1
+    module Users
+      module Auth
+        def validate_token!
+          TokenProvider.valid?(token)
+        # rescue
+        #   error!('Unauthorized', 401)
+        end
+
+        def authenticate!
+          payload, header = TokenProvider.valid?(token)
```

Comment:
There should be 2 types of authenticating methods.
1. Authenticate user. Data required: email and password via post params. Using sorcery you can do User.authenticate(email, password). Next, generate a JWT token and make a response(with the only token provided or including current user data).
2. Authenticate request. Data required: Authorization param via headers with JWT token(from 1 request). You should decode token and set @current_user variable if the user was found. Otherwise immediately return invalid token response.

Currently, I see only one type of authentication